core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -777,6 +779,69 @@ class WorkspaceServiceSpec extends AnyFlatSpec with ScalatestRouteTest with Matc
 
   }
 
+  it should "handle 404s from Sam when deleting a workspace" in withTestDataServices { services =>
+    //check that the workspace to be deleted exists
+    assertWorkspaceResult(Option(testData.workspaceNoSubmissions)) {
+      runAndWait(workspaceQuery.findByName(testData.wsName3))
+    }
+
+    when(
+      services.samDAO.deleteResource(ArgumentMatchers.eq(SamResourceTypeNames.workspace), any[String], any[UserInfo])
+    ).thenReturn(Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "404 from Sam"))))
+
+    when(
+      services.samDAO.deleteResource(ArgumentMatchers.eq(SamResourceTypeNames.workflowCollection), any[String], any[UserInfo])
+    ).thenReturn(Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "404 from Sam"))))
+
+    //delete the workspace
+    Await.result(services.workspaceService.deleteWorkspace(testData.wsName3), Duration.Inf)
+
+    verify(services.workspaceManagerDAO, Mockito.atLeast(1)).deleteWorkspace(any[UUID], any[OAuth2BearerToken])
```

Comment:
❓ Not sure if the verification has meaning in the test -- consider removing or adding an assertion around the sam calls we're expecting

Reply:
I can remove it. It was copy/paste from the simplest "delete workspace" test, so I thought it was somehow important in determining that the workspace was deleted. Lines 801-804 seem sufficient though.